R/florabank_queries.R

```diff
@@ -0,0 +1,424 @@
+#' Observations of taxa at genus level or higher are excluded. The taxonomic
+#' group can be chosen.
+#'
+#' @param connection A connection to the florabank database. See the example section
+#' for how to connect and disconnect to the database.
+#'
+#' @param begin_year Filter for observations that start from this year onwards.
+#' Default is 2010.
+#'
+#' @param ifbl_resolution The requested spatial resolution can be either
+#' 1km-by-1km IFBL squares or 4km-by-4km. Default is 1km-by-1km.
+#'
+#' @param taxongroup Choose for which taxonomic group you want the unique
+#' combinations. One of "Vaatplanten" (the default), "Mossen", "Korstmossen"
+#' of "Kranswieren".
+#'
+#' @return A dataframe with one line for each combination of taxon, IFBL-square
+#' (either at 1 km x 1 km or 4 km x 4 km resolution) and year. In case the
+#' resolution is 1 km x 1 km, a variable ifbl_4by4 gives the corresponding
+#' ifbl_4by4 identifier within which the ifbl_1by1 square is located. In case the
+#' resolution is 4 km x 4 km, the variable ifbl_squares is a concatenation of
+#' all nested squares with observations for the taxon in the corresponding year.
+#' This can be nested 1 x 1 squares as well as the corresponding 4 x 4 square
+#' (the latter is the case if the original resolution of the observation is at
+#' 4 x 4 resolution). In addition, the variable ifbl_number_squares gives the
+#' number of unique nested squares where the taxon was observed for that year
+#' and 4 x 4 square combination.
+#'
+#' @importFrom DBI dbDisconnect dbGetQuery
+#' @importFrom glue glue_sql
+#' @importFrom assertthat assert_that
+#' @importFrom stringr str_sub
+#' @importFrom dplyr %>% group_by summarize n ungroup
+#' @importFrom rlang .data
+#' @importFrom tibble as_tibble
+#'
+#' @export
+#'
+#' @examples
+#' \dontrun{
+#' # connect to florabank
+#' db_connectie <- connect_inbo_dbase("D0021_00_userFlora")
+#'
+#' # get records at 1 km x 1 km resolution for vascular plants from 2010 (default)
+#' fb_kwartier <- florabank_taxon_ifbl_year(db_connectie)
+#'
+#' # get records at 4 km x 4 km resoltion starting from 2000
+#' fb_uur <- florabank_taxon_ifbl_year(db_connectie, begin_year = 2000,
+#'  ifbl_resolution = "4km-by-4km", taxongroup = "Mossen")
+#'
+#' # disconnect from florabank
+#' dbDisconnect(db_connectie)
+#' }
+
+florabank_taxon_ifbl_year <- function(connection,
+                                      begin_year = 2010,
+                                      ifbl_resolution = c("1km-by-1km",
+                                                          "4km-by-4km"),
+                                      taxongroup = c("Vaatplanten",
+                                                     "Mossen",
+                                                     "Lichenen (korstmossen)",
+                                                     "Kranswieren")) {
+
+  assert_that(class(connection)[1] == "Microsoft SQL Server",
```

Comment:
better: `assert_that(inherits(connection), "Microsoft SQL Server")`

Reply:
Resolved in a563434